src/components/core/base/GeoLoadingHOC.tsx

```diff
@@ -0,0 +1,50 @@
+// (C) 2020 GoodData Corporation
+import * as React from "react";
+import { injectIntl } from "react-intl";
+
+import { IGeoChartInnerProps } from "../GeoChart";
+import { IntlWrapper } from "./IntlWrapper";
+import { ErrorComponent as DefaultErrorComponent } from "../../simple/ErrorComponent";
+import { ErrorStates } from "../../../constants/errorStates";
+import { generateErrorMap, IErrorMap } from "../../../helpers/errorHandlers";
+import { isLocationMissing } from "../../../helpers/geoChart";
+
+type IGeoLoadingProps = IGeoChartInnerProps;
+
+export function geoLoadingHOC<T>(InnerComponent: React.ComponentClass<T>): React.ComponentClass<T> {
+    class LoadingHOCWrapped extends React.Component<T & IGeoLoadingProps> {
```

Comment:
Please consider to
1. convert this to functional component since we don't have any state, or
2. create new `useLoading` hook, not sure how but we can give it a try :)

Reply:
we can't do that because `InnerComponent` is `ComponentClass`, so `LoadingHOCWrapped` must be a `ComponentClass`